doc/userguide/configuration/suricata-yaml.rst

```diff
@@ -1584,6 +1584,96 @@ computers etc.)
 Packet Acquisition
 ------------------
 
+Data Plane Development Kit (DPDK)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+`Data Plane Development Kit <https://www.dpdk.org/>`_ consists of libraries to accelerate packet processing
+workloads running on a wide variety of CPU architectures.
+DPDK `Environment Abstraction Layer (EAL) <https://doc.dpdk.org/guides/prog_guide/env_abstraction_layer.html>`_
+provides a generic interface to low-level resources. It is a unique way how DPDK libraries access
+NICs. EAL creates an API for application to access NIC resources from the userspace level. In DPDK, packets
+are not retrieved via interrupt handling. Instead, the application
+`polls <https://doc.dpdk.org/guides/prog_guide/poll_mode_drv.html>`_ NIC for newly received packets.
+
+EAL allows the user space application to directly access memory where NIC stores the packets.
+As a result, neither DPDK nor the application copies the packets for the inspection. The application directly
+processes packets via passed packet descriptors.
+
+
+.. figure:: suricata-yaml/dpdk.png
+    :align: center
+    :alt: DPDK basic architecture
+    :figclass: align-center
+
+    `High-level overview of DPDK application`
+
+
+Suricata makes use of DPDK for packet acquisition in workers runmode.
+The whole DPDK configuration resides in the `dpdk:` node. This node encapsulates
+2 main subnodes and those are eal-params and interfaces.
+
+::
+
+    dpdk:
+        eal-params:
+            l: 0-1
+            a: 0000:3b:00.0
+        interfaces:
+            - interface: 0000:3b:00.0 # PCIe address of the NIC port
+                # Suricata settings
+                threads: auto # or number of threads - auto takes all lcores
+                copy-mode: ips
+                copy-iface: 0000:3b:00.0 # PCIe address of the second interface
+                # NIC settings
+                promisc: true
+                multicast: true # enables also detection on multicast packets
+                checksum-checks: true # if Suricata should validate checksums
+                checksum-checks-offload: true # if possible offload checksum validation to the NIC
+                mtu: 3000 # Set MTU of the device in bytes # default 1526
+                # DPDK settings
+                mempool-size: 65356 # 64*1024
+                mempool-cache-size: 250
+                rx-descriptors: 1024
+                tx-descriptors: 1024
+
+
+The node `dpdk.eal-params` consists of `DPDK arguments <https://doc.dpdk.org/guides/linux_gsg/linux_eal_parameters.html>`_
```

Comment:
I request to recheck the logic YAML parse to embed the ports mentioned above as a whitelist. Depending upon the distro and dpdk version available option can be either '-a` or '-w`. Hence rework